brand/utils/harvest_data.py

```diff
@@ -0,0 +1,47 @@
+from datetime import datetime, timedelta
+from urllib.parse import quote
+
+from django.conf import settings
+
+import requests
+
+
+def fetch_harvest_data(brand_tag, brand_url="", brand_country="", brand_name=""):
+    base_url = "https://bank.green/harvest"
+
+    params = {
+        "bankTag": brand_tag,
+        "bankUrl": brand_url,
+        "bankCountry": brand_country,
+        "bankName": brand_name,
+    }
+
+    # URL encode the parameters
+    encoded_params = "&".join(f"{k}={quote(v)}" for k, v in params.items() if v)
+
+    url = f"{base_url}?{encoded_params}"
+
+    response = requests.get(url, headers={"Authorization": f"Token {settings.HARVEST_API_TOKEN}"})
```

Comment:
Harvest can sometimes takes 5 minutes to return results... make sure to set a request.get(timeout=600)